index.d.ts

```diff
@@ -0,0 +1,92 @@
+export class Client {
+  public config: Configuration;
+
+  constructor(apiKeyOrConfig: string | Configuration);
+
+  public notify(
+    error: Error,
+    beforeSendCallback?: BeforeSend,
+    blocking?: boolean,
+    postSendCallback?: (sent: boolean) => void,
+  ): void;
+
+  public setUser(id: string, name: string, email: string): void;
+
+  public clearUser(): void;
+
+  public leaveBreadcrumb(name: string, metadata?: IMetadata | string): void;
+}
+
+export class Configuration {
+  public version: string;
+  public apiKey?: string;
+  public delivery: IStandardDelivery;
+  public beforeSendCallbacks: BeforeSend[];
+  public notifyReleaseStages?: string[];
+  public releaseStage?: string;
+  public appVersion?: string;
+  public codeBundleId?: string;
+  public autoNotify: boolean;
+  public handlePromiseRejections: boolean;
+
+  constructor(apiKey?: string);
+
+  public shouldNotify(): boolean;
+
+  public registerBeforeSendCallback(callback: BeforeSend): void;
+
+  public unregisterBeforeSendCallback(
+    callback: BeforeSend,
+  ): void;
+
+  public clearBeforeSendCallbacks(): void;
+
+  public toJSON(): any;
+}
+
+type BeforeSend = (report: Report) => boolean | void;
+
+export class IStandardDelivery {
```

Comment:
The class is named `StandardDelivery` in your library. Since the definition should mirror the implementation itself, it's not a good idea to rename it in the definition file. If someone tries to construct this it will fail at runtime. Also capital `I` before a type name is usually reserved for interfaces (though strictly not necessary).